src/mrinufft/operators/autodiff.py

```diff
@@ -135,15 +137,25 @@ def __init__(self, nufft_op, wrt_data=True, wrt_traj=False):
         if wrt_traj and self.nufft_op.backend in ["finufft", "cufinufft"]:
             self.nufft_op.raw_op._make_plan_grad()
         self.nufft_op._grad_wrt_data = wrt_data
-
+        if wrt_traj:
+            self.samples_torch = torch.nn.Parameter(
+                data=torch.Tensor(self.nufft_op.samples),
+                requires_grad=True,
+            )
+        
     def op(self, x):
         r"""Compute the forward image -> k-space."""
-        return _NUFFT_OP.apply(x, self.samples, self.nufft_op)
+        return _NUFFT_OP.apply(x, self.samples_torch, self.nufft_op)
 
     def adj_op(self, kspace):
         r"""Compute the adjoint k-space -> image."""
-        return _NUFFT_ADJOP.apply(kspace, self.samples, self.nufft_op)
+        return _NUFFT_ADJOP.apply(kspace, self.samples_torch, self.nufft_op)
 
     def __getattr__(self, name):
         """Get the attribute from the root operator."""
-        return getattr(self.nufft_op, name)
+        if hasattr(self.nufft_op, name):
+            if name == 'samples':
+                return self.samples_torch
+            return getattr(self.nufft_op, name)
```

Comment:
Don't make the exception here, just add a property. Also this will break if you don't have it initialized. 

```python
@property
def samples(self):
    try:
        return self._samples_torch 
    except AttributeError:
        return self.nufft_op.samples

def __getattr__(self, name):
    return getattr(self.nufft_op, name)
```

Reply:
Ahhh good point, but then if we are in autodiff module with grad_wrt_traj, we SHOULD have it initialized. 
But I dont know if it misses any corner cases (perhaps if we want grad_wrt_data and request `samples`